test/integration/base.py

```diff
@@ -9,55 +9,42 @@
 ######################################################################
 from __future__ import annotations
 
-import http.client
-import os
+from test.integration.bucket_cleaner import BucketCleaner
+from test.integration.helpers import (
+    BUCKET_CREATED_AT_MILLIS,
+    random_bucket_name,
+)
 
 import pytest
 
-from b2sdk.v2 import current_time_millis
+from b2sdk.v2 import B2Api, current_time_millis
 from b2sdk.v2.exception import DuplicateBucketName
 
-from .bucket_cleaner import BucketCleaner
-from .helpers import (
-    BUCKET_CREATED_AT_MILLIS,
-    BUCKET_NAME_LENGTH,
-    GENERAL_BUCKET_NAME_PREFIX,
-    authorize,
-    bucket_name_part,
-)
-
 
+@pytest.mark.usefixtures("cls_setup")
 class IntegrationTestBase:
-    @pytest.fixture(autouse=True)
-    def set_http_debug(self):
-        if os.environ.get('B2_DEBUG_HTTP'):
-            http.client.HTTPConnection.debuglevel = 1
-
-    @pytest.fixture(autouse=True)
-    def save_settings(self, dont_cleanup_old_buckets, b2_auth_data):
-        type(self).dont_cleanup_old_buckets = dont_cleanup_old_buckets
-        type(self).b2_auth_data = b2_auth_data
-
-    @classmethod
-    def setup_class(cls):
-        cls.this_run_bucket_name_prefix = GENERAL_BUCKET_NAME_PREFIX + bucket_name_part(8)
-
-    @classmethod
-    def teardown_class(cls):
-        BucketCleaner(
-            cls.dont_cleanup_old_buckets,
-            *cls.b2_auth_data,
-            current_run_prefix=cls.this_run_bucket_name_prefix
-        ).cleanup_buckets()
+    b2_api: B2Api
+    this_run_bucket_name_prefix: str
+    bucket_cleaner: BucketCleaner
+
+    @pytest.fixture(autouse=True, scope="class")
+    def cls_setup(self, request, b2_api, b2_auth_data, bucket_name_prefix, bucket_cleaner):
+        cls = request.cls
+        cls.b2_auth_data = b2_auth_data
+        cls.this_run_bucket_name_prefix = bucket_name_prefix
+        cls.bucket_cleaner = bucket_cleaner
+        cls.b2_api = b2_api
+        cls.info = b2_api.account_info
 
     @pytest.fixture(autouse=True)
     def setup_method(self):
-        self.b2_api, self.info = authorize(self.b2_auth_data)
+        self.buckets_created = []
+        yield
+        for bucket in self.buckets_created:
+            self.bucket_cleaner.cleanup_bucket(bucket)
 
```

Comment:
1. What was wrong with `setup_class` and `teardown_class`?
2. Why not just using fixture which will create a bucket and auto-delete it afterwards? Why need to track `buckets_created`?

Reply:
1) What was wrong with setup_class and teardown_class?

I wanted to tests leveraging pytest fixtures. As you see using them with UnitTest classes is very hacky.
After enabling myself doing that, I decided I shouldn't just duplicated existing code, hence I refatored it to use more of idiomatic pytest fixtures.

2)

 some tests create multiple buckets using direct calls to self.create_bucket()

Additionally, why I try to eagerly delete buckets, instead of resulting listing them at then end:
* list-buckets operation cost money, while delete-bucket doesn't
* there is bucket limit, so eager bucket delete makes it much easier not to hit it
* with the way I refactored this, we actually call cleaner with list-buckets ONCE instead of multiple times (1 for each test class) as it was before